core/src/main/kotlin/xtdb/arrow/ArrowUtil.kt

```diff
@@ -37,4 +47,128 @@ object ArrowUtil {
             throw t
         }
     }
+
+    private val ARROW_MAGIC = "ARROW1".toByteArray(StandardCharsets.UTF_8)
+
+    /**
+     * Validates the Arrow magic bytes at the end of the buffer
+     */
+    @Throws(IllegalArgumentException::class)
+    private fun validateArrowMagic(ipcFileFormatBuffer: ArrowBuf) {
+        val capacity = ipcFileFormatBuffer.capacity()
+        val magicLength = ARROW_MAGIC.size
+
+        val isValid = ByteFunctionHelpers.compare(
+            ipcFileFormatBuffer,
+            (capacity - magicLength).toInt(),
+            capacity.toInt(),
+            ARROW_MAGIC,
+            0,
+            magicLength
+        ) == 0
+
+        if (!isValid) {
+            throw IllegalArgumentException.createNoKey("invalid Arrow IPC file format", emptyMap<String, String>())
+        }
+    }
+
+    /**
+     * Reads the Arrow footer from the given buffer
+     */
+    fun readArrowFooter(ipcFileFormatBuffer: ArrowBuf): ArrowFooter {
+        validateArrowMagic(ipcFileFormatBuffer)
+
+        val capacity = ipcFileFormatBuffer.capacity()
+        val magicLength = ARROW_MAGIC.size
+
+        // Calculate positions
+        val footerSizeOffset = capacity - (Integer.BYTES + magicLength)
+        val footerSize = ipcFileFormatBuffer.getInt(footerSizeOffset)
+        val footerPosition = footerSizeOffset - footerSize
+
+        // Get footer buffer and create Arrow Footer
+        val footerBuffer = ipcFileFormatBuffer.nioBuffer(footerPosition, footerSize)
+        return ArrowFooter(Footer.getRootAsFooter(footerBuffer))
+    }
+
+
+
+    /**
+     * Creates an Arrow record batch view from the given block and buffer
+     */
+    fun toArrowRecordBatchView(block: ArrowBlock, buffer: ArrowBuf): ArrowRecordBatch {
+        // Determine prefix size based on continuation token
+        val prefixSize = if (buffer.getInt(block.offset) == MessageSerializer.IPC_CONTINUATION_TOKEN) {
+            8
+        } else {
+            4
+        }
+
+        // Create message buffer and get record batch header
+        val messageBuffer = buffer.nioBuffer(
+            block.offset + prefixSize,
+            block.metadataLength - prefixSize
+        )
+
+        val batch  = RecordBatch()
+        Message.getRootAsMessage(messageBuffer).header(batch)
+
+        // Create and retain body buffer
+        val bodyBuffer = buffer.slice(
+            block.offset + block.metadataLength,
+            block.bodyLength
+        ).apply {
+            referenceManager.retain()
+        }
+
+        try {
+            return MessageSerializer.deserializeRecordBatch(batch, bodyBuffer)
```

Comment:
lift return out of try if possible, import-static deserializeRecordBatch

Reply:
What is the pattern then to `release` in the catch if I lift this?

Reply:
Nevermind. Misunderstood the idea.